homework/index.js

```diff
@@ -27,21 +12,151 @@
+      class: 'contributor-header',
+      text: 'Contributors',
+    });
+    const ul = createAndAppend('ul', rightContainer, {
+      class: 'contributor-list',
+    });
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-container',
+      });
+      createAndAppend('img', li, {
+        class: 'contributor-avatar',
+        src: contributor.avatar_url,
+      });
+      const contributorDataDiv = createAndAppend('div', li, {
+        class: 'contributor-data',
+      });
+      createAndAppend('a', contributorDataDiv, {
+        text: contributor.login,
+        href: contributor.html_url,
+        class: 'contributor-name',
+      });
+      createAndAppend('div', contributorDataDiv, {
+        text: contributor.contributions,
+        class: 'contributionCount',
+      });
     });
   }
 
+  async function fetchContributors(url, rightContainer) {
+    try {
+      const fetchedContributors = await fetchJSON(url);
+      renderRepoContributors(fetchedContributors, rightContainer);
+    } catch (error) {
+      createAndAppend('div', rightContainer, {
+        text: `We got an error: $ {error.message}`,
+        class: 'alert-error',
+      });
+    }
+  }
+
+  function renderRepoInfoAndContributorsOnstartup(repositories, root) {
+    createAndAppend('img', root, {
+      src: './hyf.png',
+      id: 'hyf-logo',
+      alt: 'logo image',
+    });
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    const selectMenu = createAndAppend('select', header, {
+      id: 'selectMenu',
+    });
+    const container = createAndAppend('div', root, {
+      id: 'container',
+    });
+    const leftContainer = createAndAppend('div', container, {
+      class: 'leftContainer',
+    });
+
+    const rightContainer = createAndAppend('div', container, {
+      class: 'rightContainer',
+    });
+
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
+
+    repositories.forEach((repository, i) => {
+      createAndAppend('option', selectMenu, {
+        value: i,
+        text: repository.name,
+      });
+    });
+    const firstRepository = repositories[0];
+    renderRepositoryInfo(firstRepository, leftContainer);
+    fetchContributors(firstRepository.contributors_url, rightContainer);
+
+    selectMenu.addEventListener('change', event => {
+      leftContainer.innerHTML = '';
+      renderRepositoryInfo(repositories[event.target.value], leftContainer);
+      rightContainer.innerHTML = '';
+      fetchContributors(repositories[event.target.value].contributors_url, rightContainer);
+    });
+  }
+
+  function main(url) {
+    async function fetchAllData() {
+      const root = document.getElementById('root');
+      const parsedResponse = await fetchJSON(url);
+      renderRepoInfoAndContributorsOnstartup(parsedResponse, root);
```

Comment:
When I made a network error, I got this back:
*(I removed the `u` from github url)*

<img width="730" alt="screen shot 2018-12-21 at 23 00 34" src="https://user-images.githubusercontent.com/25272822/50366903-c7f02b00-057c-11e9-80ed-12eea7b4869a.png">

what about `try {...} catch(error) {...}` blocks here ?

-------
As a taste: 😃
In my opinion `renderRepoInfoAndContributorsOnstartup` this is a bit long name as a one line,
so what about `initRepoInfoAndContributors` or may be while we are not using it for more than a one time `renderRepoInfoAndContributors`, still long but shorter 😃